commit 39aa628e17145ca250458e20ae922c3bd0e4e9c6
Author: Albert Astals Cid <aacid@kde.org>
Date:   Mon Jan 7 00:35:45 2019 +0100

    REVERT
    Remove Qt version check
    
    Now that Qt 5.10 is the minimum required version
    RJVB added the "expansion" of QQuickItem::size::isEmpty()

diff --git src/declarativeimports/core/iconitem.cpp src/declarativeimports/core/iconitem.cpp
index 2beaccdd3..66f40594f 100644
--- src/declarativeimports/core/iconitem.cpp
+++ src/declarativeimports/core/iconitem.cpp
@@ -66,6 +66,11 @@ IconItem::IconItem(QQuickItem *parent)
     connect(KIconLoader::global(), &KIconLoader::iconLoaderSettingsChanged,
             this, &IconItem::updateImplicitSize);
 
+#if QT_VERSION < QT_VERSION_CHECK(5, 10, 0)
+    connect(this, &QQuickItem::enabledChanged,
+            this, &IconItem::onEnabledChanged);
+#endif
+
     connect(this, &IconItem::implicitWidthChanged, this, &IconItem::implicitWidthChanged2);
     connect(this, &IconItem::implicitHeightChanged, this, &IconItem::implicitHeightChanged2);
 
@@ -660,8 +665,10 @@ void IconItem::itemChange(ItemChange change, const ItemChangeData &value)
 {
     if (change == ItemVisibleHasChanged && value.boolValue) {
         m_blockNextAnimation = true;
+#if QT_VERSION >= QT_VERSION_CHECK(5, 10, 0)
     } else if (change == ItemEnabledHasChanged) {
         onEnabledChanged();
+#endif
     } else if (change == ItemSceneChange && value.window) {
         if (m_window) {
             disconnect(m_window.data(), &QWindow::visibleChanged, this, &IconItem::windowVisibleChanged);
diff --git src/declarativeimports/core/windowthumbnail.cpp src/declarativeimports/core/windowthumbnail.cpp
index 64cd11e76..0835af720 100644
--- src/declarativeimports/core/windowthumbnail.cpp
+++ src/declarativeimports/core/windowthumbnail.cpp
@@ -209,8 +209,10 @@ void WindowThumbnail::itemChange(ItemChange change, const ItemChangeData &data)
         }
         break;
 
+#if QT_VERSION >= QT_VERSION_CHECK(5, 10, 0)
     case ItemEnabledHasChanged:
         Q_FALLTHROUGH();
+#endif
     case ItemVisibleHasChanged:
         if (data.boolValue) {
             if (startRedirecting()) {
diff --git src/plasmaquick/dialog.cpp src/plasmaquick/dialog.cpp
index c9f9cd87d..9aeb6c4f5 100644
--- src/plasmaquick/dialog.cpp
+++ src/plasmaquick/dialog.cpp
@@ -1208,7 +1208,7 @@ bool Dialog::event(QEvent *event)
     /*Fitt's law: if the containment has margins, and the mouse cursor clicked
      * on the mouse edge, forward the click in the containment boundaries
      */
-    if (d->mainItem && !d->mainItem->size().isEmpty()) {
+    if (d->mainItem && d->mainItem->width() > 0 && d->mainItem->height() > 0) {
         switch (event->type()) {
             case QEvent::MouseMove:
             case QEvent::MouseButtonPress:
@@ -1356,8 +1356,10 @@ void Dialog::setVisible(bool visible)
             setPosition(popupPosition(d->visualParent, size()));
         }
 
+#if QT_VERSION >= QT_VERSION_CHECK(5, 10, 0)
         // Bug 381242: Qt remembers minimize state and re-applies it when showing
         setWindowStates(windowStates() & ~Qt::WindowMinimized);
+#endif
         QQuickWindow::setVisible(visible);
         //signal will be emitted and proxied from the QQuickWindow code
     } else {
